docs: Split, update, improve the contributing guide for reviewers and committers

qmonnet opened the pull request:
- docs: Request sentence case for headings in documentation 
- docs: Break down contributing guide
- docs: Improve source formatting on review process
- docs: Consolidate review process, update periodic duties description
- docs: Add guidance for cilium/docs-structure reviews

Easier to review per commit.

Cc @aanm @margamanterola (Contributing guide & weekly duties update)
Cc @learnitall @lambdanis (docs-structure review guide)

qmonnet committed:
docs: Break down contributing guide

The Contributing guide is long. Big. Huge. Let's make it easier to read
by splitting it into three parts:

- Guidance for community contributors (remaining on the same page)
- Guidance for reviewers and committers, for the review process
- Guidance for reviewers and committers, for weekly duties

Reference the new pages from the existing one.

Apart for title formatting and new reference target, the contents in the
new page are unchanged at this stage.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: Add guidance for cilium/docs-structure reviews

Add a new document, among the guidance for reviewers/committers, to
detail the particularities and expectations for reviewing Pull Requests
as a member of the cilium/docs-structure team.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: Improve source formatting on review process

Minor formatting improvements: avoiding inline link URLs, and
re-wrapping the paragraphs, to improve readability of the source for the
document.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: Consolidate review process, update periodic duties description

Before this commit, the page on weekly duties would contain:

- A description of the PR review process duty, which describes much of
  the PR review process, and should really be in the page dedicated to
  this process

- An outdated description of the other rotations (backporter, janitor).

Let's try to improve and update this. First, we move everything related
to the review process to the dedicated page. Doing so, we update the
contents to remove any reference to a weekly duty, and to keep only the
practical aspects of the process.

Then, we rewrite the description of the weekly duties: we currently have
release manager, backporter, triager, and CI health manager. The
description for these roles is rather succinct, because I don't think
they are publicly documented anywhere at the moment, but we can improve
this in the future. We're also missing a way to identify who are the
assignees at this time.

As Joe observed, some of these duties do not rotate on a weekly basis;
so we also rename the page into "Periodic duties", and update the
description accordingly.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

learnitall commented on Documentation/contributing/development/contributing_guide.rst:
```diff
@@ -310,240 +314,8 @@ below the commit that introduced the changes and replace the work ``pick`` with
-check what the status is and establish who will work on getting the backports
-into the tree (for instance by investigating CI failures and addressing review
-feedback). Ensure that the responsibility for driving the PRs forward is clear.
+Reviewers should try to apply the documented :ref:`review_process` when
```
```suggestion
Reviewers should apply the documented :ref:`review_process` when
```

I think we could get rid of `try to` here?

learnitall commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,47 @@
+
+.. _weekly_duties:
+
+Weekly duties
```
Are we doing subheadings in caps? I think @zacharysarah started this conversation in #26632, but I don't think we finished it?

qmonnet replied:
I thought of that discussion as well and didn't fix capitalisation when I noticed it, then forgot to come back to it :sweat_smile:. I suppose the current guidance about title case still stands until we decide otherwise, so I'll fix.

learnitall commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,47 @@
+
+Backporters handle backports to Cilium's supported stable branches. The
+:ref:`backport_process` provide some guidance on how to do backports. They
+typically coordinate in #launchpad on Slack.
```
I think it would make sense to switch the last two sentences around here; couple other nits

```suggestion
Backporters handle backports to Cilium's supported stable branches. They
typically coordinate in #launchpad on Slack. The :ref:`backport_process`
provides guidance on how to do backports. 
```

learnitall commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,47 @@
+
+  - They push and merge contributions from community contributors
+  - They review updates to files without a dedicated code owner
+  - They triage bugs
```
I think it would make sense to add a note here explaining what it means to triage bugs, or what this process looks like

qmonnet replied:
It would totally make sense, but as long as the docs are not the source of truth for this role (at the moment, they're not) I don't want to go into the details and have outdated contents soon again. I can add a one-sentence summary maybe.

learnitall commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,47 @@
+
+CI Health Managers monitor the status of the CI, track down flakes, and make
+their best to ensure that CI checks keep running smoothly. They are active on
+#testing on Slack.
```
I think it would make sense to change the last sentence here to match the structure used in the last couple of sections:

```suggestion
their best to ensure that CI checks keep running smoothly. They typically coordinate in
#testing on Slack.
```

qmonnet replied:
I thought about it, although it's not coordination like for the other two. Backporters and release managers tend to announce “I'm starting a new round of backports / a new release”, making it obvious they're in charge of it, etc. By contrast, CI health folks won't do announcements, just discuss potential issues if any - so they'll typically be “active” on the channel. Although “coordinate” could probably work as well I suppose. It doesn't matter really, I'll update to your suggestion.

learnitall commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,47 @@
+CI Health Managers
+------------------
+
+CI Health Managers monitor the status of the CI, track down flakes, and make
```
```suggestion
CI Health Managers monitor the status of the CI, track down flakes, and do
```

learnitall commented on Documentation/contributing/development/reviewers_committers/index.rst:
```diff
@@ -0,0 +1,24 @@
+=======================================
+
+Some contributors have specific roles, such as reviewers or committers. The
+following resources provide guidance for some specific tasks attached to those
```
```suggestion
following resources provide guidance for specific tasks attached to those
```

qmonnet replied:
This one was intentional, to mark that not all tasks are covered. Should I remove it?

learnitall replied:
Ah I see, let's leave it in then!

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+------------------
+
+You are not expected to review the technical aspects of the documentation
+changes in a PR. Obviously, if you do have knowledge of the topic and if you
```
```suggestion
changes in a PR, however, if you do have knowledge of the topic and if you
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+Documentation structure
+-----------------------
+
+One essential part of the review is to ensure that the contribution maintains a
```
```suggestion
One essential part of a review is to ensure that the contribution maintains a
```
I feel like this needs to be `a` instead of `the`, but I'm not sure.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+consider whether the page should be split. Consider also whether new text comes
+with a satisfactory structure: does it fit well with the surrounding context,
+or did the author simply use a "note" box instead of trying to integrate the
+new information to the relevant paragraph?
```
This is a specific example of one thing to look for in a review, but it's added here in a way that implies this is the main thing that needs to be checked for.

```suggestion
with a satisfactory structure. For example, does the text fit well with the 
surrounding context, or did the author simply use a "note" box instead
of trying to integrate the new information to the relevant paragraph?
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+See :ref:`the backport criteria for documentation changes
+<backport_criteria_docs>`. Mark the PR for backports by setting the labels for
+all supported branches to which the changes apply (all supported branches
+containing the parent features to which the modified sections relate).
```
```suggestion
all supported branches to which the changes apply and contain
the parent features to which the modified sections relate.
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+CODEOWNERS updates
+~~~~~~~~~~~~~~~~~~
+
+All documentation sources are assigned for review to cilium/docs-structure by
```
Small nit

```suggestion
All documentation sources are assigned to cilium/docs-structure for review by
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+
+All documentation sources are assigned for review to cilium/docs-structure by
+default. However, when a contributor creates a new page, consider whether it
+should be covered by another team as well (for the technical aspects). If this
```
Is there a way we could remove the parentheses here?

qmonnet replied:
There's always a way :slightly_smiling_face: I'll figure out something.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+When a feature is advertised as Beta in the PR, make sure that the author
+clearly indicates it as such in the documentation, both by mentioning "(Beta)"
+in the title of the section for the feature and by including the dedicated
+banner.
```
It might be a good idea to add a note here explaining how to do this, or a copy of the text that we expect folks to include.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+
+When the PR introduces new user-facing options, metrics, or behavior, in such a
+way that it affects upgrades or downgrades, ensure that the author summarizes
+the changes with a note in Documentation/operations/upgrade.rst.
```
```suggestion
the changes with a note in `Documentation/operations/upgrade.rst`.
```

Does this need backticks?

qmonnet replied:
Double back-ticks, even.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+should be automatically updated when parts of the Cilium repository change. You
+should ensure that everything is up-to-date, although the CI does in fact
+handle the verification for you. But you may have to accompany contributors and
+help them figure out the command they need to run to perform the update.
```
There are a couple of items in this paragraph that we discuss but don't provide specific definitions or pointers for. I think a user may have trouble understanding what exactly is meant here. For instance, I'm left with the following questions:

* What is a reference document used for?
* How can I identify a reference document?
* What changes to the Cilium repository require my attention in this matter?
* What do I need to do to ensure everything is up to date?
* How do I know what command I need to run to perform the update? Where can I find it?

qmonnet replied:
This would make for an entire separate guide, and I'm sure we'd struggle to keep it up-to-date :confused:. Not sure how to address this.

learnitall replied:
Maybe we could add it as a todo and just leave this be for now?

qmonnet replied:
Yes, I'll open an issue for the build framework and mention this as well.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Documentation checks include running a spell checker, which has a list of
+exceptions ("good words"). Team cilium/docs-structure is the owner for this
```
```suggestion
exceptions, referred to as "good words". Team cilium/docs-structure is the owner for this
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+
+The Documentation checks include running a spell checker, which has a list of
+exceptions ("good words"). Team cilium/docs-structure is the owner for this
+file. There's usually not much to look at, but it's good to keep it clean:
```
This is great voice but may not fit in a technical document.

learnitall replied:
Also see [1]

qmonnet replied:
What do you mean by “this is great voice”? I'm not sure I understand, I can rephrase to be more suited to a tech doc.

qmonnet replied:
updated

learnitall replied:
Ah sorry I wasn't more clear on this- the changes you made look great. By "this is great voice" I meant that it had great personality. But from my understanding docs need to be more dry.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+exceptions ("good words"). Team cilium/docs-structure is the owner for this
+file. There's usually not much to look at, but it's good to keep it clean:
+
+  - Entries should remain sorted alphabetically (mind the case)
```
I think it would make sense to remove the word `should` here and also remove the parentheses.

```suggestion
  - Entries are sorted alphabetically. Words starting with uppercase letters come before words starting with lowercase letters.
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+  - Entries should remain sorted alphabetically (mind the case)
+  - If a new entry is misspelled, reject it
+  - Lowercase entries are considered case-insensitive, so it's no use adding an
+    entry with capital letters if the lowercase version is already in
```
This line also has some great voice, but it may not fit. [1]

qmonnet replied:
Can you please explain? I can change the “it's no use”, is this what you're talking about?

qmonnet replied:
updated

learnitall replied:
The changes you made look great! Sorry I wasn't clear- it was the "it's no use" that I was referring to.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+file. There's usually not much to look at, but it's good to keep it clean:
+
+  - Entries should remain sorted alphabetically (mind the case)
+  - If a new entry is misspelled, reject it
```
I think this could be reworded to make the 'Entry' more of a subject.
```suggestion
  - Entries in the wordlist should be spelled correctly.
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+Netlify preview
+~~~~~~~~~~~~~~~
+
+Netlify builds a new preview for each PR touching the documentation. You are
```
Can we add a link to Netlify's homepage or the like?

qmonnet replied:
I can, but I think you can't access much if you haven't been added to the right Netlify team, so the interest is limited.

What really matters here would be the links to the Netlify previews (or build failures) in the list of GitHub workflows, I can maybe try to mention that.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+Netlify builds a new preview for each PR touching the documentation. You are
+not expected to check the preview for each PR. However, do have a look if the
+PR contains some involved formatting changes, such as nested blocks or
+directives, or work on tables or tabs. Also look at the preview if you have a
```
```suggestion
directives, or changes to tables or tabs. Also look at the preview if you have a
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+Ask the contributors to rework any sentence that is too difficult to read or to
+understand.
+
+We aim at keeping the documentation clean, consistent, and in a clear and
```
It might make sense here to remove `we` and replace it with `sig/docs-structure`.

qmonnet replied:
Right, this was better suited for the Google Doc than for Cilium's docs. Good catch.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+potential style improvements can be time consuming, so the amount of effort
+that you put into style guidance is up to you.
+
+As of this writing, there is no tooling in place to enforce particular style
```
Should we rely on versioning to handle in-time statements?

```suggestion
There is no tooling in place to enforce particular style
```

learnitall replied:
Also see [2]

qmonnet replied:
No that's another leftover from the Google Doc, I should just switch to your suggestion.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+The build framework
+-------------------
+
+To do. See:
```
To do :)

qmonnet replied:
Yep, I was kinda hoping I could get away with this one. I don't have this content yet and wasn't sure it was worth blocking this guide for it. I know we just started to discourage their use in the docs though :confused:

learnitall replied:
Maybe we could compromise and add this as a todo in an issue, but remove the `To do` line in the docs?

qmonnet replied:
Sounds good. I'll open a separate issue for this, and for the auto-generated docs.

qmonnet replied:
https://github.com/cilium/cilium/issues/27098

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+Documentation changes trigger the build of a new Netlify preview. If the build
+fails, the PR authors or reviewers must investigate it (ideally the author, but
+in practice, contributors are not always familiar with RST or with our build
+framework, so consider giving a hand).
```
Can we remove the parentheses here?

qmonnet replied:
Certainly

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+Checkpatch
+~~~~~~~~~~
+
+The Checkpatch workflow (part of the BPF tests) is not directly relevant to
```
```suggestion
The Checkpatch workflow is part of the BPF tests and is not directly relevant to
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+~~~~~~~~~~
+
+The Checkpatch workflow (part of the BPF tests) is not directly relevant to
+documentation, but may raise some patch formatting issues, so you should run it
```
It might make sense to add information on what a patch formatting issue is, but definitely not necessary.

qmonnet replied:
It's “commit subject is too long”.

... unless you start touching the `bpf/` code.

I'll figure out something.

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+~~~~~~~~~~~~~~~~~
+
+Integration tests (on Travis, or on GitHub Actions) are the only workflow that
+rebuilds the ``docs-builder`` image. Building this image is necessary to
```
```suggestion
Integration tests run on Travis and GitHub Actions are the only workflows that
rebuild the ``docs-builder`` image. Building this image is necessary to
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+Integration tests (on Travis, or on GitHub Actions) are the only workflow that
+rebuilds the ``docs-builder`` image. Building this image is necessary to
+validate changes to the ``Documentation/Dockerfile`` or to the list of Python
+dependencies (from ``Documentation/requirements.txt``). The GitHub workflow
```
```suggestion
dependencies found in ``Documentation/requirements.txt``. The GitHub workflow
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+
+Integration tests also run a full build in the Cilium repository, including the
+post-build checks (and in particular, ``Documentation/Makefile``'s ``check``
+target). Therefore, they are able to raise inconsistencies in auto-generated
```
```suggestion
post-build checks, in particular, ``Documentation/Makefile``'s ``check``
target. Therefore, integration tests are able to raise inconsistencies in auto-generated
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+Ready to merge
+--------------
+
+Committers usually require a PR to pass the full suite of CI test before they
```
```suggestion
Committers usually require a PR to pass the full suite of CI tests before they
```

learnitall commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,228 @@
+--------------
+
+Committers usually require a PR to pass the full suite of CI test before they
+can merge it. However, as of this writing, there is no clear consensus on
```
[2]

learnitall reviewed (changes requested):
Thank you for this @qmonnet :heart:! This looks great! There are a couple of places that I think need some more background information to help users, but other than that I just have a bunch of nits on sentence structure, word choice, and voice. Also, IMO I think we should try to avoid using parentheses when possible, and there are a lot of parentheses used in these changes, so I made some suggestions on rephrasing things to remove them.

margamanterola reviewed (approved):
LGTM with regards to rotations and structure. I did a quick read of the whole text and didn't find any issues to highlight.

learnitall commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,49 @@
+  - They review updates to files without a dedicated code owner
+  - They triage bugs, which means they interact with reporters until the issue
+    is clear and can get the label associated to the corresponding working
+    group, when possible.
```
Nit:
```suggestion
    group, when possible
```

learnitall reviewed (approved):
Awesome looks great! Thank you!

zacharysarah commented on Documentation/contributing/development/contributing_guide.rst:
```diff
@@ -10,6 +10,10 @@
 
+This document explains how to contribute as a community contributor.
+:ref:`Guidance for reviewers and committers <reviewer_committer>` is also
+available.
```
Subtle wording edits
```suggestion
This document shows how to contribute as part of the community.
:ref:`Guidance for reviewers and committers <reviewer_committer>` is also
available.
```

qmonnet replied:
I'd like to keep “as a community contributor“, as I think it better highlights the distinction between the different roles, and in particular the reviewer and committer roles mentioned right after.

zacharysarah replied:
That makes sense.

zacharysarah commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,49 @@
+
+.. _weekly_duties:
+
+Weekly Duties
```
Sentence case in headings? 🥺  
```suggestion
Weekly Duties
```

qmonnet replied:
OK, but then let's make it “official”. I'll add a commit to change the directive in the style guide.

zacharysarah commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,49 @@
+=============
+
+Some members of the Cilium organization will have rotational duties that change
+every week.
```
Use present tense
```suggestion
Some members of the Cilium organization have rotational duties that change
every week.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,49 @@
+----------------
+
+Release managers take care of the patch releases for each supported stable
+branch of Cilium. They typically coordinate in #launchpad on Slack.
```
A link would be good here, or more specificity
```suggestion
branch of Cilium. They typically coordinate in #launchpad on Cilium Slack.
```

qmonnet replied:
I realise links to Cilium Slack throughout the docs are a mess (I count 13 occurrences of the URL). I'll harmonise them in a follow-up PR.

qmonnet replied:
https://github.com/cilium/cilium/pull/27346

zacharysarah commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,49 @@
+
+Backporters handle backports to Cilium's supported stable branches. They
+typically coordinate in #launchpad on Slack. The :ref:`backport_process`
+provide some guidance on how to do backports.
```
Specificity re:Slack, verb tense agreement, clear plain language
```suggestion
typically coordinate in #launchpad on Cilium Slack. The :ref:`backport_process`
provides guidance on how to backport.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -0,0 +1,49 @@
+
+CI Health Managers monitor the status of the CI, track down flakes, and do
+their best to ensure that CI checks keep running smoothly. They typically
+coordinate in #testing on Slack.
```
Clear plain language, specificity re:Slack
```suggestion
CI Health Managers monitor the status of the CI, track down flakes, and
ensure that CI checks keep running smoothly. They typically
coordinate in #testing on Cilium Slack.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/index.rst:
```diff
@@ -0,0 +1,24 @@
+Contributing as a Reviewer or Committer
+=======================================
+
+Some contributors have specific roles, such as reviewers or committers. The
+following resources provide guidance for some specific tasks attached to those
+roles. Refer to `Cilium's Contributor Ladder`_ for details
+about the different roles.
```
Well written ✨

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+Cilium contributors who are responsible for maintaining the good state of the
+project's documentation, by reviewing Pull Requests (PRs) that update the
+documentation. Each time a non-draft PR touching files owned by the team is
+opened on GitHub, one member of the team is automatically assigned for review.
```
Assignment is a GitHub Action, right? So we can use active voice accurately here?
```suggestion
documentation. Each time a non-draft PR touching files owned by the team opens on GitHub, a GitHub Action automatically assigns one member of the team for review.
```

qmonnet replied:
It's GitHub code review auto-assignment system, I don't think it's an Action itself (my understanding is that an Action is a block that can be used to build Workflows; and that “GitHub Actions” also designate the whole automation system?). But yeah, I can just use “GitHub” as a subject.

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+
+You are not expected to review the technical aspects of the documentation
+changes in a PR, however, if you do have knowledge of the topic and if you find
+some elements that are incorrect or missing, do flag them.
```
```suggestion
You are not expected to review the technical aspects of the documentation
changes in a PR. However, if you do have knowledge of the topic and if you find
some elements that are incorrect or missing, do flag them.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+coherent structure for the documentation. Ask yourself if the changes are
+located on the right page, at the right place. This is especially important if
+pages are added, removed, or shuffled around. If the addition is large,
+consider whether the page should be split. Consider also whether new text comes
```
wording nit
```suggestion
consider whether the page needs to split. Consider also whether new text comes
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+------------------------------
+
+Backports Labels
+~~~~~~~~~~~~~~~~
```
typo nit
```suggestion
Backport Labels
~~~~~~~~~~~~~~~
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+~~~~~~~~~~~~~~~
+
+When a feature is advertised as Beta in the PR, make sure that the author
+clearly indicates it as such in the documentation, both by mentioning "(Beta)"
```
Edits for clarity
```suggestion
clearly indicates the Beta state in the documentation, both by mentioning "(Beta)"
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+clearly indicates it as such in the documentation, both by mentioning "(Beta)"
+in the title of the section for the feature and by including the dedicated
+banner. Users can include this banner by using the ``include`` directive for
+the ``Documentation/beta.rst`` fragment.
```
It would be good to include an example here:
```suggestion
the ``Documentation/beta.rst`` fragment.

.. parsed-literal::
    
    .. include:: Documentation/beta.rst
```

qmonnet replied:
I didn't want initially because the relative path depends on where we include from, and I didn't want to mislead the readers; but it's true we can just as well use an absolute path.

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+
+When the PR introduces new user-facing options, metrics, or behavior, in such a
+way that it affects upgrades or downgrades, ensure that the author summarizes
+the changes with a note in ``Documentation/operations/upgrade.rst``.
```
Clarity
```suggestion
When the PR introduces new user-facing options, metrics, or behavior
that affects upgrades or downgrades, ensure that the author summarizes
the changes with a note in ``Documentation/operations/upgrade.rst``.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+should be automatically updated when parts of the Cilium repository change. You
+should ensure that everything is up-to-date, although the CI does in fact
+handle the verification for you. But you may have to accompany contributors and
+help them figure out the command they need to run to perform the update.
```
Edits for clarity
```suggestion
Some reference documents (command reference, Helm reference) 
are automatically updated when parts of the Cilium repository change. You
Although the CI handles verification, you can still ensure that everything is
up-to-date, although the CI. You may have to help contributors
figure out which command they need to perform the update.
```

qmonnet replied:
I rewrote the whole paragraph:

```
When certain parts of the Cilium repository change, contributors may have to
update some auto-generated reference documents that are part of Cilium's
documentation, such as the command reference or the Helm reference. The CI
validates that these updates are present in the PR. If they are missing, you
may have to help contributors figure out what commands they need to run to
perform the updates. These commands are usually provided in the logs of the
GitHub workflows that failed to pass.
```

zacharysarah replied:
Even better. ✨

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+uses a file containing a list of exceptions to ignore, to make it possible to
+use technical terms that are not in its dictionary. Team cilium/docs-structure
+is the owner for this file. Usually, there is not much feedback to provide on
+updates to this file, but reviewers should keep the following items in mind:
```
Edits for clarity
```suggestion
updates to this file. It's useful for reviewers to know that:
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Documentation checks include running a spell checker. This spell checker
+uses a file containing a list of exceptions to ignore, to make it possible to
```
```suggestion
uses a file, ``Documentation/spelling_wordlist.txt``, containing a list of spelling exceptions to ignore.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+
+The Documentation checks include running a spell checker. This spell checker
+uses a file containing a list of exceptions to ignore, to make it possible to
+use technical terms that are not in its dictionary. Team cilium/docs-structure
```
```suggestion
Team cilium/docs-structure
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+The Documentation checks include running a spell checker. This spell checker
+uses a file containing a list of exceptions to ignore, to make it possible to
+use technical terms that are not in its dictionary. Team cilium/docs-structure
+is the owner for this file. Usually, there is not much feedback to provide on
```
```suggestion
owns this file and its associated script, ``Documentation/update-spelling_wordlist.sh``. Usually, there is not much feedback to provide on
```

qmonnet replied:
True, but I'm not sure what this info brings in the context of the instructions for reviewing updates to the spelling exceptions :thinking:. I'd rather have it mentioned as part of the docs build framework, when we describe it?

zacharysarah replied:
That's fair; the script doesn't need a mention here.

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+  - Entries are sorted alphabetically, with all words starting with uppercase
+    letters coming before words starting with lowercase letters.
+  - Entries in the list of exceptions must be spelled correctly.
+  - Lowercase entries are considered case-insensitive, so reviewers should
```
```suggestion
  - Lowercase entries are case-insensitive, so reviewers should
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+`Netlify`_ builds a new preview for each PR touching the documentation. You are
+not expected to check the preview for each PR. However, do have a look if the
+PR contains some involved formatting changes, such as nested blocks or
+directives, or changes to tables or tabs. Also look at the preview if you have
```
```suggestion
not expected to check the preview for each PR. However, if the
PR contains detailed formatting changes, such as nested blocks or
directives, or changes to tables or tabs, it's good to verify that changes
render as expected. Also verify the preview if you have
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+author uses.
+
+The link to the Netlify preview is available from the list of the checks on the
+page of the PR. If the preview build failed, the link leads to the build logs.
```
```suggestion
The list of checks on the PR page contains a link to the Netlify preview.
If the preview build failed, the link leads to the build logs.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+Read :ref:`Cilium's documentation style guide <docs_style_guide>`.
+
+Flag poor formatting or obvious mistakes. The syntax for RST is not always
+trivial and some contributors get it wrong, or they simply forget this is RST
```
```suggestion
trivial and some contributors make mistakes, or they simply forget to use RST
```

qmonnet replied:
Clearly sounds better, thanks

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+
+Flag poor formatting or obvious mistakes. The syntax for RST is not always
+trivial and some contributors get it wrong, or they simply forget this is RST
+and they employ Markdown mark-up instead. Make sure they fix such issues.
```
```suggestion
and they employ Markdown mark-up instead. Make sure authors fix such issues.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+substitutions, and if so, do they use the right directive? If not, do they use
+the right language?
+
+Beyond that, the amount of time you should spend on suggestions for improving
```
```suggestion
Beyond that, the amount of time you spend on suggestions for improving
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+Read :ref:`Cilium's documentation style guide <docs_style_guide>`.
+
+Flag obvious grammar mistakes. Try to read the updated text as a user would.
+Ask the contributors to rework any sentence that is too difficult to read or to
```
```suggestion
Ask the contributors to revise any sentence that is too difficult to read or to
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+Ask the contributors to rework any sentence that is too difficult to read or to
+understand.
+
+@cilium/docs-structure aims at keeping the documentation clean, consistent, and
```
```suggestion
@cilium/docs-structure aims to keep the documentation clean, consistent, and
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+
+@cilium/docs-structure aims at keeping the documentation clean, consistent, and
+in a clear and comprehensible state. User experience must always be as good as
+possible. To achieve this objective, Documentation updates should follow best
```
```suggestion
possible. To achieve this objective, Documentation updates must follow best
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,240 @@
+possible. To achieve this objective, Documentation updates should follow best
+practices, such as the ones from the style guide, but reviewing PRs in-depth to
+flag all potential style improvements can be time consuming, so the amount of
+effort that you put into style guidance is up to you.
```
```suggestion
practices, such as the ones from the style guide. Reviewing PRs at sufficient depth to
flag all potential style improvements can be time consuming, so the amount of
effort you put into style guidance is up to you.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+
+.. note::
+
+   These instructions assume that whoever is reviewing is a member of the
```
```suggestion
   These instructions assume that reviewers are members of the
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+.. note::
+
+   These instructions assume that whoever is reviewing is a member of the
+   Cilium GitHub organization or has the status of a committer. This is
```
```suggestion
   Cilium GitHub organization or have the status of a committer. This is
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+`teams <cilium_teams>`_ should review the PR using the ``CODEOWNERS`` file.
+Each committer can see the PRs they need to review by filtering by reviews
+requested. A good filter is provided in this `link <user_review_filter_>`_ so
+make sure to bookmark it.
```
```suggestion
Once a contributor opens a PR, GitHub automatically picks which
`teams <cilium_teams>`_ should review the PR using the ``CODEOWNERS`` file.
Each committer can see the PRs they need to review by filtering by reviews
requested. A good filter is provided in this `link <user_review_filter_>`_ so
make sure to bookmark it.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+review the entire PR. However, if the PR is quite large then it can help to
+narrow the area of focus to one particular aspect of the code. When leaving a
+review, share which areas you focused on and which areas you think that other
+reviewers should look into. This will help others to focus on aspects of review
```
```suggestion
reviewers should look into. This helps others focus on aspects of review
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+reviewers should look into. This will help others to focus on aspects of review
+that have not been covered as deeply.
+
+Belonging to a team does not mean that a committer should know every single
```
```suggestion
Belonging to a team does not mean that a committer needs to know every single
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+that have not been covered as deeply.
+
+Belonging to a team does not mean that a committer should know every single
+line of code the team is maintaining. For this reason it is recommended that
```
```suggestion
line of code the team is maintaining.
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+
+Belonging to a team does not mean that a committer should know every single
+line of code the team is maintaining. For this reason it is recommended that
+once you have reviewed a PR, if you feel that another pair of eyes is needed,
```
```suggestion
Once you review a PR, if you feel that another pair of eyes is needed,
```

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+Belonging to a team does not mean that a committer should know every single
+line of code the team is maintaining. For this reason it is recommended that
+once you have reviewed a PR, if you feel that another pair of eyes is needed,
+you should re-request a review from the appropriate team. In the example below,
```
```suggestion
it's okay to re-request a review from the appropriate team. In the following example,
```

qmonnet replied:
Actually... Let's just instruct to re-request a review, “it's okay to” is not necessary. We do want more reviews if the reviewers are unsure.

zacharysarah commented on Documentation/contributing/development/reviewers_committers/review_process.rst:
```diff
@@ -0,0 +1,168 @@
+   :scale: 50%
+
+When all review objectives for all ``CODEOWNERS`` are met, all required CI
+tests have passed and a proper release label as been set, you may set the
```
```suggestion
tests have passed and a proper release label is set, you may set the
```

joestringer replied:
We can follow up on this in a subsequent change so that it draws appropriate review, but recently we've had discussions that reviewers should not set `ready-to-merge` as MLH will handle that. Otherwise it's up to either tophats or committers to notice the PRs and merge them.

zacharysarah reviewed (changes requested):
@qmonnet Thank you so much for this! I have some wording suggestions and another plea to use sentence casing for headings. 🙏🏻

qmonnet committed:
docs: Request sentence case for headings in documentation

As discussed by some documentation reviewers, update the style guide to
instruct contributors to use sentence case rather than title case on
headings in documentation.

We don't update the whole documentation, which has never been consistent
in that regard.

However, we do update the guidance for writing documentation itself, as
these pages should set an example for the rest of the docs, and should
follow the guidance they set themselves.

Suggested-by: Sarah Corleissen <sarah.corleissen@isovalent.com>
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

joestringer commented on Documentation/contributing/development/reviewers_committers/duties.rst:
```diff
@@ -9,182 +9,41 @@
-submit changes to these steps if you have found a better way to perform each
-duty.
+Some members of the Cilium organization have rotational duties that change
+every week.
```
nit: Release Managers don't change every week. Maybe `periodically` instead?

joestringer commented on Documentation/contributing/development/reviewers_committers/review_docs.rst:
```diff
@@ -0,0 +1,249 @@
+Keep an eye on :ref:`code-blocks <docs_style_code_blocks>`: do they include RST
+substitutions, and if so, do they use the right directive? If not, do they use
+the right language?
+
```
Not sure whether it's specifically worth mentioning here or if we think it's covered by "read the style guide" but callouts are another low-hanging fruit that is worth keeping vigilant over. I think that this is pretty well understood across @cilium/docs-structure already today, but if we think it's helpful to also explicitly mention here then by all means let's do so. I don't have a strong opinion about this.

qmonnet replied:
Callouts are also explicitly mentioned on this page, but earlier, under "Documentation structure" (given that they're not just formatting, but more a matter of how the new info is incorporated into the document).

joestringer reviewed (approved):
LGTM, couple of minor nits and one potential content change that should maybe be handled separately to avoid stacking a subtle process change into this PR which is otherwise general improvements.

zacharysarah reviewed (approved):
Thanks for the updates, @qmonnet!
